pkg/manager/manager_test.go

```diff
@@ -296,6 +296,15 @@ var _ = Describe("manger.Manager", func() {
 				<-m2done
 			})
 
+			It("should return an error if it can't create a ResourceLock", func() {
```

Comment:
Please add a testcase where you call `manager.New` with any non-default resourcelock and verify that you end up getting that. You can use the existing `should default to ConfigMapsLeasesResourceLock` testcase to get some inspiration

Reply:
Makes sense, done. Extended the existing case to test all resource lock types. PTAL :)